src/executor_config/barrier.md

```diff
@@ -0,0 +1,89 @@
+-TODO: All Barriers Explained
+-TODO: Security Note about misconfigured barriers
+-TODO: Example
+
+# Barrier
+Before any XCMs are executed in the XCM executor, they must pass the Barrier.
+The Barrier type implements the `ShouldExecute`
+[trait](https://paritytech.github.io/polkadot/doc/xcm_executor/traits/trait.ShouldExecute.html)
```

Comment:
Given all these links being so long, it might be useful to list them at the bottom of the file. Something like: 

```
some [`link`] foo


[`link`]: https://paritytech.github.io/polkadot/doc/xcm_executor/traits/trait.ShouldExecute.html
```

This works in rust-docs, I think it is a native markdown feature.

Reply:
Did not know this! Really nice feature.